File: EJERCICIO_1/EJERCICIOSPHP/rouster.php
<?php
class Rouster {
    private $calculatedValue;

    public function __construct($totalBonus, $totalDiscount, SalaryBass $salary) {
        $this->calculatedValue = $totalBonus - $totalDiscount + $salary->calculateSalary();  // Assuming calculateSalary exists in SalaryBase
    }

    public function calculateRouster() {
        return $this->calculatedValue;
    }
}
?>

File: EJERCICIO_1/EJERCICIOJAVA/rouster.java
package EJERCICIOJAVA;

public class rouster {
    private double calculate;

    public rouster(double totalBonus,  double totaldiscount, salaryBass salary) {

        this.calculate = totalBonus - totaldiscount + salary.calculateSalary();  //Assuming calculateSalary exists in salaryBass
    }

    public double calculateRouster() {
        return this.calculate;
    }
}